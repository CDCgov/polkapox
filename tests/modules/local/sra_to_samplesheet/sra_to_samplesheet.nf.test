nextflow_process {

    name "Test Process SRA_TO_SAMPLESHEET"
    script "modules/local/sra_to_samplesheet/sra_to_samplesheet.nf"
    process "SRA_TO_SAMPLESHEET"

    test("Create samplesheet from SRA") {

        when {
            params {
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = [meta: "test_sub",  fastq1: file("${projectDir}/tests/nf-test-data/test_sub_1.fq.gz")]
                input[1] = [meta: "test_sub",  fastq2: file("${projectDir}/tests/nf-test-data/test_sub_2.fq.gz")]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
