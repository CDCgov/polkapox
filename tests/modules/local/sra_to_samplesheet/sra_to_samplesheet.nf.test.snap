{
    "Create samplesheet from SRA": {
        "content": [
            {
                "0": [
                    [
                        "test_sub",
                        "samplesheet.csv:md5,ee90b888f2f5d5a82745270c4d2a1aff"
                    ]
                ],
                "samplesheet": [
                    [
                        "test_sub",
                        "samplesheet.csv:md5,ee90b888f2f5d5a82745270c4d2a1aff"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-07-11T16:11:24.789345868"
    }
}